src/main/java/com/jlr/ttl/ds/api/controllers/LocationController.java

```diff
@@ -0,0 +1,40 @@
+package com.jlr.ttl.ds.api.controllers;
+
+import com.jlr.ttl.ds.api.dto.entity.Location;
+import com.jlr.ttl.ds.api.dto.entity.Vehicle;
+import com.jlr.ttl.ds.api.repositories.LocationRepository;
+import com.jlr.ttl.ds.api.repositories.VehicleRepository;
+import com.jlr.ttl.ds.api.services.LocationService;
+import com.jlr.ttl.ds.api.services.VehicleService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping(path = "/api/load_management/locations")
+@CrossOrigin("http://localhost:4200")
+public class LocationController {
+
+    @Autowired
+    private LocationService locationService;
+    @Autowired
+    private LocationRepository locationRepository;
+
+    @GetMapping("/all")
+    public ResponseEntity<List<Location>> getAllLocations() {
+        List<Location> response = locationService.getAllLocationService();
```

Comment:
Please review the error handling in both endpoints for this controller. While the current code is functional it can be improved. Please implement an error handling more in line with how DSUserController does it